lib/order.rb

```diff
@@ -0,0 +1,22 @@
+require_relative 'menu'
+
+class Order
+
+  attr_reader :current_order
+
+  def initialize(menu)
+    @menu = menu
+    @current_order = Hash.new(0)
+    @total = 0
+  end
+
+  def add(dish, quantity)
+    raise 'Item is not available' if @menu.dishes[dish] == nil
```

Comment:
@menu.dishes[dish] -> what this does it goes into your the instance of the menu class you passed into order, it looks at the hash you created e.g.{ spring_rolls: 5} and it asks for the value that the key (dish) exposes. So if the item isn't in the hash then the value will be nil